doc/command_line/field_processing.md

```diff
@@ -0,0 +1,126 @@
+## Field Processing
+
+Ruby supports <i>field processing</i>.
+
+This means that when certain command-line options are given,
+the invoked Ruby program can process input line-by-line.
+
+### About the Examples
+
+Examples here assume that this code (creating file `desiderata.txt`)
+has been executed:
+
+```
+text = <<EOT
+Go placidly amid the noise and the haste,
+and remember what peace there may be in silence.
+As far as possible, without surrender,
+be on good terms with all persons.
+EOT
+File.write('desiderata.txt', text)
+```
+
+The examples also use command-line option `-e`,
+which passes the Ruby code to be executed on the command line itself:
+
+```sh
+$ ruby -e 'puts "Hello, World."'
+```
+
+### Option `-n`
+
+Option `-n` runs your program in a Kernel#gets loop:
+
+```
+while gets
+  # Your Ruby code.
+end
+```
+
+Note that `gets` reads the next line and sets global variable `$_`
+to the last read line:
+
+```sh
+$ ruby -n -e 'puts $_' desiderata.txt
+Go placidly amid the noise and the haste,
+and remember what peace there may be in silence.
+As far as possible, without surrender,
+be on good terms with all persons.
+```
+
+### Option `-p`
+
+Option `-p` is like option `-n`, but also prints each line:
+
+```sh
+$ ruby -p -e 'puts $_.size' desiderata.txt
+42
+Go placidly amid the noise and the haste,
+49
+and remember what peace there may be in silence.
+39
+As far as possible, without surrender,
+35
+be on good terms with all persons.
+```
+
+### Option `-a`
+
+Option `-a`, when given with either of options `-n` or `-p`,
+splits the string at `$_` into an array of strings at `$F`:
+
+```sh
+$ ruby -an -e 'p $F' desiderata.txt
+["Go", "placidly", "amid", "the", "noise", "and", "the", "haste,"]
+["and", "remember", "what", "peace", "there", "may", "be", "in", "silence."]
+["As", "far", "as", "possible,", "without", "surrender,"]
+["be", "on", "good", "terms", "with", "all", "persons."]
+```
+
+The default field separator used for the splitting is the input field separator `$;`.
+
+### Option `-F`
+
+Option `-F`, when given with option `-a`,
+specifies that its argument is to be the input field separator to be used for splitting:
+
+```sh
+$ ruby -an -Fs -e 'p $F' desiderata.txt
+["Go placidly amid the noi", "e and the ha", "te,\n"]
+["and remember what peace there may be in ", "ilence.\n"]
+["A", " far a", " po", "", "ible, without ", "urrender,\n"]
+["be on good term", " with all per", "on", ".\n"]
+```
+
+The argument may be a regular expression;
+see String#split.
```

Comment:
I'm not sure how to specify a regular expression here. I tried `-F/the/` or `-F\s` and none of them worked. Can you explain?

Reply:
I was trying to accommodate the comment at https://github.com/ruby/ruby/pull/10008#discussion_r1493670245.  I'm not sure exactly what we should say.

Reply:
I'll have an update for this soon.

Reply:
Fixed (added @nobu example and remark).